### PR TITLE
Check for nil pointer with StorageClass name

### PR DESCRIPTION
**Pull request by jarrpa**
Signed-off-by: Jose A. Rivera <jarrpa@redhat.com>

### CODE REVIEW

File: pkg/controller/storagecluster/reconcile.go [hunk 1]
```diff
@@ -917,11 +917,16 @@ func newStorageClassDeviceSets(sc *ocsv1.StorageCluster) []rook.StorageClassDevi
...
+	if ds.DataPVCTemplate.Spec.StorageClassName == nil || *ds.DataPVCTemplate.Spec.StorageClassName == "" {
+		return false, fmt.Errorf("no StorageClass specified")
+	}
+	storageClassName := *ds.DataPVCTemplate.Spec.StorageClassName
+
 	storageClass := &storagev1.StorageClass{}
 	err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: "", Name: storageClassName}, storageClass)
 	if err != nil {
-		return false, fmt.Errorf("failed to retrieve Storage Class %q. %+v", storageClassName, err)
+		return false, fmt.Errorf("failed to retrieve StorageClass %q. %+v", storageClassName, err)
```
Comment (obnoxxx) [inline pkg/controller/storagecluster/reconcile.go]: Unrelated spelling change. Please exclude from this patch.

```suggestion
		return false, fmt.Errorf("failed to retrieve Storage Class %q. %+v", storageClassName, err)
```

File: pkg/controller/storagecluster/reconcile.go [hunk 2]
```diff
@@ -917,11 +917,16 @@ func newStorageClassDeviceSets(sc *ocsv1.StorageCluster) []rook.StorageClassDevi
 	return storageClassDeviceSets
 }
 
-func (r *ReconcileStorageCluster) throttleStorageDevices(storageClassName string) (bool, error) {
+func (r *ReconcileStorageCluster) throttleStorageDevices(ds ocsv1.StorageDeviceSet) (bool, error) {
```
Comment (obnoxxx) [inline pkg/controller/storagecluster/reconcile.go]: imho, it is a bit strange to have this check in the `throttleStorageDevices` function. It should be a different function or directly implemented in `ensureCephCluster`!?

**Comment by obnoxxx** [comment]
@jarrpa  here is what I meant with not putting the check into the throttle function but into a separate function: https://github.com/obnoxxx/ocs-operator/commits/validateStorageDeviceSet

**Comment by jarrpa** [comment]
Review comments addressed. Split validation into its own function and chopped up the commit.

**Comment by obnoxxx** [comment]
https://github.com/openshift/ocs-operator/pull/449 has the spelling fix now

**Comment by jarrpa** [comment]
Spelling change commit removed.
